client/state/ui/layout-focus/reducer.js

```diff
@@ -0,0 +1,42 @@
+/**
+ * External dependencies
+ */
+import includes from 'lodash/includes';
+
+/**
+ * Internal dependencies
+ */
+import {
+	LAYOUT_FOCUS_SET,
+	LAYOUT_NEXT_FOCUS_ACTIVATE,
+	LAYOUT_NEXT_FOCUS_SET,
+} from 'state/action-types';
+
+const initialState = { current: null, previous: null, next: null };
+const validAreas = [ 'content', 'sidebar', 'sites', 'preview' ];
+const isValidArea = area => includes( validAreas, area );
+
+export default function layoutFocus( state = initialState, action ) {
+	switch ( action.type ) {
+		case LAYOUT_FOCUS_SET:
+			if ( action.area === state.current || ! isValidArea( action.area ) ) {
+				return state;
+			}
+			return Object.assign( {}, state, { current: action.area, previous: state.current } );
+		case LAYOUT_NEXT_FOCUS_SET:
+			if ( action.area === state.next || ! isValidArea( action.area ) ) {
+				return state;
+			}
+			return Object.assign( {}, state, { next: action.area } );
+		case LAYOUT_NEXT_FOCUS_ACTIVATE:
+			// If we don't have a change queued and the focus has changed
+			// previously, set it to `content`. This avoids having to set the
+			// focus to content on all navigation links because it becomes the
+			// default after focus has shifted.
+			if ( ! state.next && ! state.previous ) {
```

Comment:
I don't fully understand the design (this behavior and comment were copied directly from the old module) but I *think* what this is for is when Calypso first loads, we automatically switch to `content` layout-focus. `previous` would thus be being used to determine if this is a new page load.

In any case, layoutFocus is so crucial to Calypso's functionality that I didn't want to risk breaking anything while we moved it over, which is why I left the functionality the same. Personally I'd rather improve the code by making additional PRs against the reducer rather than trying to do it all at once, but others may disagree.